untrusted/urts/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+///
+/// Avoid storing the de-referenced instance of the enclave.  The de-referenced
+/// value of the enclave will result in failures to the SGX SDK after the
+/// enclave is dropped.
+#[derive(Debug, PartialEq)]
+pub struct Enclave {
+    // The enclave ID, assigned by the SGX interface
+    id: sgx_enclave_id_t,
+}
+
+/// Build an [Enclave] for use with SGX calls.
+pub struct EnclaveBuilder {
+    // The bytes for the enclave.
+    bytes: Vec<u8>,
+
+    // `true` if the enclave should be created in debug mode
+    debug: bool,
+}
+
+impl EnclaveBuilder {
+    /// Returns an EnclaveBuilder for the provided signed enclave.
+    ///
+    /// # Arguments
+    ///
+    /// * `bytes` - The bytes representing the enclave file.  This should be a
+    ///     signed enclave.
+    pub fn new(bytes: &[u8]) -> EnclaveBuilder {
+        EnclaveBuilder {
+            bytes: bytes.into(),
+            debug: false,
+        }
+    }
+
+    /// Toggle debugging of the enclave on or off.  The default is off.
+    ///
+    /// # Arguments
+    ///
+    /// * `debug` - `true` to enable enclave debugging, `false` to disable it.
+    #[must_use]
+    pub fn debug(mut self, debug: bool) -> EnclaveBuilder {
+        self.debug = debug;
+        self
+    }
+
+    /// Create the enclave
+    ///
+    /// Will talk to the SGX SDK to create the enclave.  Once the enclave has
+    /// been created then calls into the enclave can be made by de-referencing
+    /// the enclave.
+    ///
+    /// See
+    /// <https://download.01.org/intel-sgx/sgx-dcap/1.13/linux/docs/Intel_SGX_Enclave_Common_Loader_API_Reference.pdf>
+    /// for error codes and their meaning.
+    pub fn create(mut self) -> Result<Enclave, Error> {
+        let mut enclave_id: sgx_enclave_id_t = 0;
+        let result = unsafe {
+            // Per the API reference `buffer` is an input, however the signature
+            // lacks the const qualifier.  Through testing it has been shown
+            // that `sgx_create_enclave_from_buffer_ex()` *will* modify the
+            // `buffer` parameter.  This can be seen by copying the input bytes
+            // and comparing before and after.
+            //
+            //      let mut buffer = self.bytes.to_vec();
+            //      println!("Pre comparing {}", buffer.as_slice() == self.bytes);
+            //      let result = unsafe {sgx_create_enclave_from_buffer_ex(...)};
+            //      println!("Post comparing {}", buffer.as_slice() == self.bytes);
+            //
+            // The modification that `sgx_create_enclave_from_buffer_ex()`
+            // makes to the `buffer` is such that if one were to re-use the
+            // modified buffer in another call to
+            // `sgx_create_enclave_from_buffer_ex()` then
+            // `SGX_ERROR_INVALID_ENCLAVE_ID` would be returned.
+            sgx_create_enclave_from_buffer_ex(
+                self.bytes.as_mut_ptr(),
+                self.bytes.len().try_into().unwrap(),
```

Comment:
This may have been my rebasing/restacking on the core types, but this was already addressed.
Mainly want to be sure the correct diff is being reviewed 
The comment shows "outdated"
![image](https://user-images.githubusercontent.com/101757619/178814790-0676e1ce-074c-45fd-b83d-20036d374868.png)